### PR TITLE
list log drains by DB (from addon id)

### LINKED ISSUE #571: Add log drains to DB using addon_id
User will provide `addon <addon_uuid>` to the `log-drains` command.
Example:
```bash
$ scalingo --app my-app --addon ad-1234567-1123 log-drains-add \
--type papertrail --host logs.papertrailapp.com --port 10303
```

`POST /apps/:app_id/addons/:addon_id/log_drains`
Payload for the API:
```json
{
  "drain": {
    "type": "papertrail",
    "host": "logs.papertrailapp.com",
    "port": "10303"
  }
}
```

### PR DESCRIPTION
**Pull request by curzolapierre**
Command: list log drains by DB (from addon uuid)

Example:
```bash
$ scalingo --app my-app log-drains --addon ad-9be0fc04-bee6-4981-a403-a9ddbee6bd1f
+----------------+---------------+------------------------------------------+
|   ADDON NAME   |  ADDON PLAN   |                 URL                      |
+----------------+---------------+------------------------------------------+
| Scalingo Redis | redis-sandbox | tcp+tls://logs.papertrailapp.com:10303   |
|                |               | ovh://:id@tag1.logs.ovh.com:6514         |
+----------------+---------------+------------------------------------------+

$ scalingo --app my-app log-drains --with-addon
+-----------------+------------------+----------------------------------------+
|   ADDON NAME    |    ADDON PLAN    |                 URL                    |
+-----------------+------------------+----------------------------------------+
| Scalingo Redis  | redis-sandbox    | tcp+tls://logs.papertrailapp.com:10303 |
+                 +                  +----------------------------------------+
|                 |                  | ovh://:id@tag1.logs.ovh.com:6514       |
+-----------------+------------------+----------------------------------------+
| Scalingo Mongo  | mongo-sandbox    | ovh://:id@tag1.logs.ovh.com:6514       |
+-----------------+------------------+----------------------------------------+
```

Fix #571

### CODE REVIEW

File: cmd/log_drains.go [hunk 1]
```diff
@@ -12,11 +12,20 @@ var (
 	logDrainsListCommand = cli.Command{
 		Name:     "log-drains",
 		Category: "Log drains",
-		Flags:    []cli.Flag{appFlag},
-		Usage:    "List the log drains of an application",
+		Flags: []cli.Flag{appFlag,
+			addonFlag,
+			cli.BoolFlag{Name: "with-addon", Usage: "also list the log drains of all addons"},
```
Comment (EtienneM) [inline cmd/log_drains.go:17]:
```suggestion
			cli.BoolFlag{Name: "with-addons", Usage: "also list the log drains of all addons"},
```

File: cmd/log_drains.go [hunk 2]
```diff
@@ -27,7 +36,19 @@ var (
 				return
 			}
 
-			err := log_drains.List(currentApp)
+			var addonID string
+			if c.GlobalString("addon") != "<addon_id>" {
```
Comment (EtienneM) [inline cmd/log_drains.go:40]: There is already a few commands which have a `addon` flag. Please have a look at how they handle it. For instance in the file `cmd/databases.go` on line 40.

File: cmd/log_drains.go [hunk 3]
```diff
@@ -27,7 +36,19 @@ var (
...
+				addonID = c.GlobalString("addon")
+			} else if c.String("addon") != "<addon_id>" {
+				addonID = c.String("addon")
+			}
+
+			var err error
+			if addonID != "" || c.String("with-addon") == "true" {
```
Comment (EtienneM) [inline cmd/log_drains.go:47]: `with-addon` being a `BoolFlag`, you should get its value with `Bool`:

```suggestion
			if addonID != "" || c.Bool("with-addon") {
```

**Comment by curzolapierre** [comment]
Add separation lines could be usefull in case of multiple addons + application in the table:
```bash
+-------------------------------+--------------------------------------------------------------------+
|             NAME              |                                URL                                 |
+-------------------------------+--------------------------------------------------------------------+
| sample-go-martini             | ovh://:e8ea17bd-34e5-47b6-a016-675077053417@gra3.logs.ovh.com:6514 |
+                               +--------------------------------------------------------------------+
|                               | tcp+tls://logs.papertrailapp.com:10303                             |
+-------------------------------+--------------------------------------------------------------------+
| Scalingo Redis                | ovh://:e8ea17bd-34e5-47b6-a016-675077053417@gra3.logs.ovh.com:6514 |
+                               +--------------------------------------------------------------------+
|                               | tcp+tls://logs.papertrailapp.com:10303                             |
+-------------------------------+--------------------------------------------------------------------+
```
And disable it in case of just one listing
```bash
+-------------------------------+--------------------------------------------------------------------+
|             NAME              |                                URL                                 |
+-------------------------------+--------------------------------------------------------------------+
| Scalingo Redis                | ovh://:e8ea17bd-34e5-47b6-a016-675077053417@gra3.logs.ovh.com:6514 |
|                               | tcp+tls://logs.papertrailapp.com:10303                             |
+-------------------------------+--------------------------------------------------------------------+

```

Case without separation lines:
```bash
+-------------------------------+--------------------------------------------------------------------+
|             NAME              |                                URL                                 |
+-------------------------------+--------------------------------------------------------------------+
| sample-go-martini             | ovh://:e8ea17bd-34e5-47b6-a016-675077053417@gra3.logs.ovh.com:6514 |
|                               | tcp+tls://logs.papertrailapp.com:10303                             |
| Scalingo Redis                | ovh://:e8ea17bd-34e5-47b6-a016-675077053417@gra3.logs.ovh.com:6514 |
|                               | tcp+tls://logs.papertrailapp.com:10303                             |
+-------------------------------+--------------------------------------------------------------------+
```

File: cmd/log_drains.go (hunk 2 above)
Comment (curzolapierre) [reply cmd/log_drains.go:40]: We cannot use the same method because `addonName` makes the `addon_id` mandatory

File: log_drains/list_addon.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	AddonName string
+}
+
+func ListAddon(app string, addonID string) error {
+	c, err := config.ScalingoClient()
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
+	}
+
+	resources, err := c.AddonsList(app)
```
Comment (EtienneM) [inline log_drains/list_addon.go:23]:
```suggestion
	addons, err := c.AddonsList(app)
```

`resources` does not look like a good variable name, don't you think?

File: log_drains/list_addon.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/olekukonko/tablewriter"
+	"gopkg.in/errgo.v1"
+)
+
+type addonObject struct {
+	Drains    []scalingo.LogDrain
+	AddonName string
+}
+
+func ListAddon(app string, addonID string) error {
```
Comment (EtienneM) [inline log_drains/list_addon.go:17]: I think I would prefer a different signature like:

```suggestion
type ListAddonOpts struct {
   WithAddons bool
   AddonID string
}

func ListAddon(app string, opts ListAddonOpts) error {
```

Comment (EtienneM) [inline log_drains/list_addon.go:17]: The more I think about it, the more I think you should not add a new function but use the `List` you already wrote. There is a lot of code in common! What do you think?

Comment (curzolapierre) [reply log_drains/list_addon.go:17]: I was afraid it was gonna turn into a big if else, in addition of the condition of with-addon / with id.
But we do find the list code into the list_addon code and we have also duplication in the `cmd/log_drains.go` file too.

**Comment by curzolapierre** [comment]
I would prefer your example but it seems `tablewriter` doesn't offer the possibility to do it.

**Comment by curzolapierre** [comment]
The 6th example is what I implemented. It uses the combination of `table.SetRowLine(true)` with `table.SetAutoMergeCells(true)`
so in our case the input looks like: 
```bash
+-------------------------------+--------------------------------------------------------------------+
|             NAME              |                                URL                                 |
+-------------------------------+--------------------------------------------------------------------+
| sample-go-martini             | ovh://:e8ea17bd-34e5-47b6-a016-675077053417@gra3.logs.ovh.com:6514 |
+                               +--------------------------------------------------------------------+
|                               | tcp+tls://logs.papertrailapp.com:10303                             |
+-------------------------------+--------------------------------------------------------------------+
| Scalingo Redis                | ovh://:e8ea17bd-34e5-47b6-a016-675077053417@gra3.logs.ovh.com:6514 |
+                               +--------------------------------------------------------------------+
|                               | tcp+tls://logs.papertrailapp.com:10303                             |
+-------------------------------+--------------------------------------------------------------------+
```

File: Gopkg.toml [hunk 6]
```diff
@@ -26,7 +26,7 @@
 
 [[constraint]]
   name = "github.com/Scalingo/go-scalingo"
-   version = "^4"
+  branch = "feature/cli/571/list_log_drains_by_DB"
```
Comment (EtienneM) [inline Gopkg.toml:29]: It seems the go-scalingo PR is not merged yet.

**Pull request merged**